runtime/gc_vlhgc/RememberedSetCard.hpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 1991, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup gc_vlhgc
+ */
+
+#if !defined(REMEMBEREDSETCARD_HPP)
+#define REMEMBEREDSETCARD_HPP
+
+#include "modron.h"
+
+class MM_RememberedSetCard
```

Comment:
This is cleaner, but I getting greedy and I'll suggest more (but if you feel it's excessive requirement, no problem - we can go with this as-is and I can re-eval my thinking at a later point with fresh brain)

I'd like to see a field uintptr_t/uint_32 (same as you had with typedef orignal) inside this class. Field name could be 'address' (or 'value'). We would have a constructor and allow instantiation. We could even have  getter/setter to access address field, although it's probably excessive - the field could just simple be a public.

Then,
UDATA card = 0;
would turn into
Card card(0);

readCard would return _instance_ of Card, and since  operator '=' would work, code like

UDATA card = readCard(); would revert back to

Card card = readCard(...).

Passing card as parameter (of type Card) would also work since there is default copy operator.

Operator == is unfortunately not defined by default so it should be defined or a method isEqual should be introduced.

Various spots in the code that do arithmetic operations under card will have to access it with
card.address rather than just using card. For example class methods substruct/add would than have to do something like:

(uint32_t*)base.address + index

Finally, the 4 methods (all except cardSize()) could even turn into non-statics.

Reply:
This all sounds good, but let's leave it for another PR - I'm worried that introducing object wrappers may degrade the performance of the standard builds. I'm pretty confident these changes will not.

If you're generally happy, I'll whip this into shape and squash it for final review next week.

Reply:
Sounds good. I'll have one more pass early next week.